packages/store/docs/store-taxonomy.md

```diff
@@ -0,0 +1,67 @@
+
+All stores are Far objects (a kind of remotable) that may be snapshotted to (or restored from) a corresponding pass-by-copy objects. MapStores snapshot into CopyMaps. SetStores snapshot into CopySets.
+
+# API Differences
+
+## SetStores vs MapStores*
+
+* SetStores store only keys (of type Key), where each key is key-unique, i.e., it is not keyEQ to any other key in the set.
+* MapStores store key value associations, where the keys are key-unique Keys, as with sets, and the associated values are Passable.
+
+## Weak vs Strong*
+
+* A weak store does not enable its contents to be iterated. Only with the key of a given entry will a weak store reveal anything about the entry. This sometimes enables gc of entries that the collector knows can never be mentioned. But, unlike JavaScript WeakSets and WeakMaps, a weak store *may* allow other values as keys, such as numbers, that can be synthesized at will. Entries indexed by such synthesizable keys can never be collected.
+* A strong store does enable iteration of its contents, and so must retain those contents as long as the strong store itself exists.
+
+# Storage and Representation Differences
+
+## Scalar Keys Only* vs Composite Keys Allowed
+
+* Scalars are primitives or remotables, those keys that can be compared without looking inside objects. This directly reflects the semantics of JavaScript's Maps and Sets, which only index by scalar keys.
+* Composite keys are things like copyArrays, copyRecords, copySets, or copyMaps, that compare by structural key equality.
+
+## Unstructured* vs Schematized
+
+* An unstructued store is restricted only by the representation limitations listed here. They keys must be Keys. The values must be Passable. Scalar stores allow only scalar keys. Persistent stores only allow contents that are well defined after vat death.
+* A schematized store is further restricted by a pattern provided when it was created to serve as its schema. A schematized store will only allow contents that match its schema pattern.
+
+## Long* vs Short Expected Lifetime (WeakStores only)
+
+The expected lifetime of a weak store helps us optimize its representation.
+* If we expect a weak store to outlive most of its keys, we may internally use a JavaScript WeakMap.
+* If we expect the keys in a weak store to outlive the store, we may internally use a JavaScript Map, to avoid the costs often associated indexing a key for JavaScript WeakMaps that are no longer alive. Though keep in mind that this representation denies the garbage collector the opportinity to collect those keys while these expected-to-be-short-lived stores are still alive.
+
+## Heap* vs Virtual
```

Comment:
Should this now be
```suggestion
## Small* vs Big
```
?

Reply:
I vote no.  "Big" is a good modifier to use in the names of the collection makers and their associated collection types, but big and small are very confusing modifiers to use in prose when discussing on-the-heap vs. swap-to-disk distinction.  In particular, we have virtual objects (as distinct from regular, on-the-heap JavaScript objects) and it would be really odd to refer to them as "big objects" and regular JS objects as "small objects".  You'd use virtual objects in a case where you expect to have a lot of instances, not because the instances themselves are big.  "Big" works when talking about a "big collection" because it captures the idea that the collection can grow to a very large number of elements (i.e., it can get big) because it's not limited by heap space, not because the elements themselves are individually large.

Reply:
Good point that "big object" does not work at all!

I look forward to seeing what naming system you settle on. Please let me know if you want to discuss first.

Reply:
I already signed up to wordsmith some other stuff in this document, so I'll do a writing pass and then there'll be revised prose for you to comment on when I push the next commit.